### PR TITLE
rgw: admin api - add ability to sync user stats from admin api

### PR DESCRIPTION
**Pull request by nathanejohnson**
This adds the ability to trigger synchronizing user stats from the admin api.

http://tracker.ceph.com/issues/21301

### CODE REVIEW

**Comment by joscollin** [comment]
@nathanejohnson Please see: See this too: https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#1-sign-your-work

**Comment by nathanejohnson** [comment]
@joscollin Sorry about that, I amended the commit message.  Let me know if this is sufficient.

Thank you

**Comment by nathanejohnson** [comment]
@joscollin I have tagged my commit.  By the way, with the tags do you differentiate between a bug fix versus a new feature with the tags, or do they all prefix with Fixes: ?

So regarding #5 in the guide, there is no documentation that I could find for the stats flag, but I would be happy to add it if you could point me in the right direction.

**Comment by joscollin** [comment]
@nathanejohnson It appears now, sorry.

**Comment by nathanejohnson** [comment]
@joscollin regarding documentation of the sync parameter, should I add that to this PR?  I can document the stats flag as well, but would that be more appropriate in a separate PR?

Thanks

**Comment by joscollin** [comment]
@nathanejohnson It would be appropriate in a separate PR, which mentions this PR.

File: src/rgw/rgw_rest_user.cc [hunk 1]
```diff
@@ -47,8 +48,11 @@ void RGWOp_User_Info::execute()
 
   RESTArgs::get_bool(s, "stats", false, &fetch_stats);
 
+  RESTArgs::get_bool(s, "sync", false, &sync_stats);
+
```
Comment (joscollin) [inline src/rgw/rgw_rest_user.cc]: `--sync-stats` seems already exists in rgw_admin. So could you please clarify how do you use the command now ?

The tracker shows: `$ radosgw-admin user stats --uid=<uid> --sync-stats`

Comment (nathanejohnson) [reply src/rgw/rgw_rest_user.cc]: If you call this command without --sync-stats , the stats may be stale.  I noticed this when running some of the integration tests on an admin api client library I wrote.  i.e., I could copy a file to a bucket, but stats would return with all zeros.  All this PR would do is expose that same functionality via the admin API.

Comment (joscollin) [reply src/rgw/rgw_rest_user.cc]: Looks good. But I would request a review from @yehudasa too.

**Comment by joscollin** [comment]
Test Result:

- http://pulpito.front.sepia.ceph.com/jcollin-2017-09-25_04:39:54-rgw-wip-jcollin-testing_09-25-2017-distro-basic-smithi/
- http://pulpito.front.sepia.ceph.com/compare/?branch=wip-jcollin-testing_09-25-2017&suite=rgw

**Comment by joscollin** [comment]
@nathanejohnson What about the `doc` updates ? I think this PR will wait until the doc changes are completed.

**Comment by nathanejohnson** [comment]
@joscollin I'll get to those asap

**Comment by mattbenjamin** [comment]
@joscollin results look clean modulo known valgrind issues and apparently unrelated ovh issues

**Pull request merged**